src/Deku/Interpret.js

```diff
@@ -90,47 +254,57 @@ export const makeText_ = (runOnJust) => (tryHydration) => (maybe) => (a) => (sta
 				.item(0))
 		) {
 			var i = 0;
-			// if the length is one, that means that there is only a comment because
-			// the text is empty
-			// so we need to add a text node
-			if (dom.childNodes.length === 1) {
-				dom.prepend(document.createTextNode(""));
-			} else {
-				for (var i = 0; i < dom.childNodes.length; i++) {
-					if (
-						dom.childNodes[i].nodeType === 8 &&
-						dom.childNodes[i].nodeValue === ptr
-					) {
-						i = i - 1;
-						break;
+			for (; i < dom.childNodes.length; i++) {
+				const ptrSplit = ptr.split('@-@');
+				if (
+					dom.childNodes[i].nodeType === 8 &&
+					dom.childNodes[i].nodeValue === ptrSplit[0]
+				) {
+					i = i - 1;
+					var textWasBlank = i === -1;
+					var textWasBlankAfterDynBeacon = dom.childNodes[i].nodeType === 8;
+					if (textWasBlank) {
+						dom.prepend(document.createTextNode(""));
+					}
+					if (textWasBlankAfterDynBeacon) {
+						dom.insertBefore(document.createTextNode(""), dom.childNodes[i + 1]);
 					}
+					break;
 				}
 			}
+			const main = dom.childNodes[i];
 			state.units[ptr] = {
 				// if we've done ssr for a text node, it will be a span,
 				// so we want to get the child node
-				main: dom.childNodes[i],
+				main,
+				pos: a.pos,
 				parent: a.parent,
 				scope: a.scope,
 			};
+			state.revUnits[main] = ptr;
 			return true;
 		}
 		return false;
 	})();
 	if (!iRan) {
+		const main = document.createTextNode("");
 		state.units[ptr] = {
-			main: document.createTextNode(""),
+			main,
 			parent: a.parent,
 			scope: a.scope,
+			pos: a.pos,
+			dynFamily: a.dynFamily
 		};
-		connectXToY_(maybe, ptr, a.parent, state);
+		state.revUnits[main] = ptr;
```

Comment:
This puts the string `[object Text]` into `state.revUnits`??

Reply:
Fixed.